diff --git a/node_modules/@actions/http-client/lib/index.js b/node_modules/@actions/http-client/lib/index.js
index 5a3cad3..363604a 100644
--- a/node_modules/@actions/http-client/lib/index.js
+++ b/node_modules/@actions/http-client/lib/index.js
@@ -502,7 +502,7 @@ class HttpClient {
         if (this._keepAlive && useProxy) {
             agent = this._proxyAgent;
         }
-        if (this._keepAlive && !useProxy) {
+        if (!useProxy) {
             agent = this._agent;
         }
         // if agent is already assigned use that agent.
@@ -535,15 +535,16 @@ class HttpClient {
             this._proxyAgent = agent;
         }
         // if reusing agent across request and tunneling agent isn't assigned create a new agent
-        if (this._keepAlive && !agent) {
+        // Always use a custom agent so that `keep-alive` works correctly
+        // on both `node16` and `node20` runtimes. See:
+        // - https://github.com/node-fetch/node-fetch/issues/1735
+        // - https://github.com/nodejs/node/issues/37184
+        // - https://github.com/nodejs/node/pull/43522
+        if (!agent) {
             const options = { keepAlive: this._keepAlive, maxSockets };
             agent = usingSsl ? new https.Agent(options) : new http.Agent(options);
             this._agent = agent;
         }
-        // if not using private agent and tunnel agent isn't setup then use global agent
-        if (!agent) {
-            agent = usingSsl ? https.globalAgent : http.globalAgent;
-        }
         if (usingSsl && this._ignoreSslError) {
             // we don't want to set NODE_TLS_REJECT_UNAUTHORIZED=0 since that will affect request for entire process
             // http.RequestOptions doesn't expose a way to modify RequestOptions.agent.options
